### PR TITLE
:electron: Migrations out of sync modal

### PR DESCRIPTION
**Pull request by MikesGlitch**
# When will this display?

When you have multiple clients using one server. One client has been updated, the other hasn't. The updated client runs on the budget and migrates the data to the latest. Then the out-of-date client opens the budget.

We shouldn't let the user continue using the client in case they end up with bad data. It's not worth the risk. We should warn the user about it, prompt them to update their client, and let them know it's possible to use the web app from app.actualbudget.org.

image

## Original intention
This was intended to be a **warning for when server is a different version from client**. I couldn't do it because 
- we'd need to force the user to be online and connect to the server to get the version info. 
- the user would have to wait until the server responds before loading the budget. 

I don't think we should do that so I had to bin the idea.

### CODE REVIEW

File: packages/desktop-client/src/components/Modals.tsx [hunk 1]
```diff
@@ -591,15 +592,8 @@ export function Modals() {
           return <ImportYNAB5Modal key={name} />;
         case 'import-actual':
           return <ImportActualModal key={name} />;
-        case 'manager-load-backup':
```
Comment (MikesGlitch) [inline packages/desktop-client/src/components/Modals.tsx:594]: This isn't used anywhere, cleaning it up

File: packages/loot-core/src/client/actions/budgets.ts [hunk 2]
```diff
@@ -57,7 +57,9 @@ export function loadBudget(id: string, options = {}) {
 
     if (error) {
       const message = getSyncError(error, id);
-      if (error === 'out-of-sync-migrations' || error === 'out-of-sync-data') {
+      if (error === 'out-of-sync-migrations') {
+        dispatch(pushModal('out-of-sync-migrations'));
+      } else if (error === 'out-of-sync-data') {
```
[the remark this answers is not in the document]
Comment (MikesGlitch) [reply packages/loot-core/src/client/actions/budgets.ts:62]: How has it been removed? Can you see it in the else statement?

File: packages/desktop-client/src/components/modals/OutOfSyncMigrationsModal.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  const closeBudgetAndModal = (close: () => void) => {
+    dispatch(closeBudget());
+    close();
+  };
+
+  return (
+    <Modal name="out-of-sync-migrations">
+      {({ state: { close } }) => (
+        <>
+          <ModalHeader title={<ModalTitle title="Please update Actual!" />} />
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/modals/OutOfSyncMigrationsModal.tsx:26]: **🥜 nitpick:** ‏
```suggestion
          <ModalHeader title={<ModalTitle title={t('Please update Actual!')} />} />
```

File: packages/desktop-client/src/components/modals/OutOfSyncMigrationsModal.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+              maxWidth: 550,
+              lineHeight: '1.5em',
+            }}
+          >
+            <Text>
+              <Paragraph style={{ fontSize: 16 }}>
+                It looks like you&apos;re using an outdated version of the
+                Actual. Your budget data has been updated by another client, but
+                this client is still on the old verison. For the best
+                experience, please update Actual to the latest version.
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/modals/OutOfSyncMigrationsModal.tsx:42]: **🥜 nitpick:** ‏`Trans`

File: upcoming-release-notes/3600.md [hunk 5]
```diff
@@ -0,0 +1,6 @@
+---
+category: Maintenance
```
Comment (MatissJanis) [inline upcoming-release-notes/3600.md:2]: **🥜 nitpick:** ‏
```suggestion
category: Enhancements
```

**Review by MatissJanis — APPROVED** [review]
Thanks for this! I hope I didn't put you in a `Trans` 🤯 😄

**Review by MatissJanis — APPROVED** [review]

**Pull request merged**